coalib/output/ConsoleInteraction.py

```diff
@@ -164,35 +158,47 @@ def print_lines(console_printer,
     :param sourcerange:     The SourceRange object referring to the related
                             lines to print.
     """
+
     for i in range(sourcerange.start.line, sourcerange.end.line + 1):
         console_printer.print(format_lines(lines='', line_nr=i),
                               color=FILE_LINES_COLOR,
                               end='')
+
         line = file_dict[sourcerange.file][i - 1].rstrip("\n")
-        tab_width = int(section.get('tab_width', 4))
-        s = SpacingHelper(tab_width)
-        tab_dict = dict(s.yield_tab_lengths(line))
+        # Get suitable lexer depending on filename's extension and if not found
+        # fallback to TextLexer().
+        try:
+            lexer = get_lexer_for_filename(sourcerange.file)
+        except ClassNotFound:
+            lexer = TextLexer()
+        s = SpacingHelper.DEFAULT_TAB_WIDTH
+        lexer.add_filter(VisibleWhitespaceFilter(spaces="•", tabs=True,
+                                                 tabsize=s))
+        # highlight() combines lexer and formatter to output a ``str``
+        # object.
         printed_chars = 0
         if i == sourcerange.start.line and sourcerange.start.column:
-            print_spaces_tabs_in_unicode(
-                console_printer, line[:sourcerange.start.column-1],
-                tab_dict, FILE_LINES_COLOR)
+            console_printer.print(highlight(
+                 line[:sourcerange.start.column-1],
+                 lexer, TerminalTrueColorFormatter()).rstrip("\n"), end='')
 
             printed_chars = sourcerange.start.column-1
 
         if i == sourcerange.end.line and sourcerange.end.column:
-            print_spaces_tabs_in_unicode(
-                console_printer, line[printed_chars:sourcerange.end.column-1],
-                tab_dict, HIGHLIGHTED_CODE_COLOR, printed_chars)
-
-            print_spaces_tabs_in_unicode(
-                console_printer, line[sourcerange.end.column-1:],
-                tab_dict, FILE_LINES_COLOR, sourcerange.end.column-1)
+            console_printer.print(highlight(
+                line[printed_chars:sourcerange.end.column-1],
+                lexer, TerminalTrueColorFormatter(style=BackgroundSourceRange))
+                .rstrip("\n"), end='')
```

Comment:
Do you mean `line[:-1]` ? It's not correct, will show repeated characters. But I'm not sure what do you actually mean ?